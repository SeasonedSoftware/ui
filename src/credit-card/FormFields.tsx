import React, { Fragment } from 'react'

import Input from '../form/Input'

export default ({ fields, formState: {values}, handleFocus }:any) => {
  return (
    <Fragment>
      <Input
        {...fields.text('name')}
        key="name"
        onFocus={handleFocus}
        value={values.name}
        autoFocus
      />
      <Input {...fields.text('number')} key="number" onFocus={handleFocus} />
      <Input
        {...fields.text('expiry')}
        key="expiry"
        placeholder={'MM/AA'}
        onFocus={handleFocus}
        value={values.expiry}
      />
      <Input
        {...fields.text('cvc')}
        label="CVC"
        placeholder={'123'}
        onFocus={handleFocus}
        value={values.cvc || ''}
      />
    </Fragment>
  )
}
