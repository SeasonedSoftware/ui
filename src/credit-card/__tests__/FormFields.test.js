import React from 'react'
import renderer from 'react-test-renderer'

import FormFields from '../FormFields'

jest.mock('../../form/Input', () => props => <div {...props}>Input</div>)

it('renders correctly', () => {
  const props = {
    fields: {
      text: jest.fn(() => ({})),
    },
    formState: {
      values: {},
    },
    handleFocus: jest.fn(),
  }

  const tree = renderer.create(<FormFields {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
