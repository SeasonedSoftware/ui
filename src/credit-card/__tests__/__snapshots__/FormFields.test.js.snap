// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    autoFocus={true}
    onFocus={[MockFunction]}
  >
    Input
  </div>,
  <div
    onFocus={[MockFunction]}
  >
    Input
  </div>,
  <div
    onFocus={[MockFunction]}
    placeholder="MM/AA"
  >
    Input
  </div>,
  <div
    label="CVC"
    onFocus={[MockFunction]}
    placeholder="123"
    value=""
  >
    Input
  </div>,
]
`;
